### DISCUSSION TITLE
issue: potencial bug: setValue does render updates of fields controlled by useFieldArray

### DISCUSSION THREAD

**Discussion by prztrz** [react-hook-form/react-hook-form#8033] [Q&A]
### Version Number

7.28

### Codesandbox/Expo snack

https://codesandbox.io/s/boring-snow-of7b3o?file=/src/App.tsx

### Steps to reproduce

I have a from data structure which contains nested array:

```ts
type Pet = {
  name: string;
  species: string;
};
type Person = {
  name: string;
  pets: Pet[];
};
export type FormValues = {
  people: Person[];
};
```

When I want to use `setValue` to set a new value for `people` array (the only difference is the new value have `people.0.pets` array elements reordered), the value is getting set (which is confirmed via log inside `changeValue` method). But `Pets` component using `useFieldsArray` hook. Still displays inputs in old order.

In my real life application I use form context and controller for inputs so i implemented them in sandbox replication example for consistency.

### Expected behaviour

I expect when `setValue` changes the value which contains changed nested arrays, theese arrays should be re rendered by components using `useFieldArray`

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
please use the latest version, this should fix the issue. an important note here is this `setValue` with `useFieldArray` will be deprecated in the next major version. I would consider using `useFieldArray` methods instead. 

you can refer to the rules section: https://react-hook-form.com/api/useform/setvalue#rules

**Comment by prztrz** [comment]
hi @bluebill1049 thank ypu for such a quick answer! 

Could you tell me specificaly which rule did I break?  I cannot really use `useFieldArray` as my Array are much deeper in react tree than drag and drop context in which I reorder the array elements. For now I found that `setValue` targeting speciffic nested array instead to targeting its parent object does the job.

```ts
// this works
const reordered = [getValues('people.0.pets.1'), getValues('people.0.pets.0')]
setValue('people.0.pets', reordered)

// this nope
setValue(people, [{...getValues('people.0', pets: reordered}, getValues(people.1)]
```
is my approach correct?

**Comment by bluebill1049** [comment]
you didn't break any rules, i am just suggesting to move away to use setValue to update filed array, your issue is working when you upgrade to the latest version, but action as such probably should use `replace` from `useFieldArray` instead.

**Reply by prztrz** [reply]
Hmm But I use v `7.28.0` already, and I believe  this is the latest version, and the problem still occur.

**Reply by bluebill1049** [reply]
https://user-images.githubusercontent.com/10513364/158901818-530e4972-9d94-4b5d-9976-53a5459c55ce.mov

maybe i missing something here, feel free to point out above.

I did with the latest and works fine, perhaps you want to try with `replace` from `useFieldArray` instead.
